src/transport/raw/Base.h

```diff
@@ -44,18 +45,14 @@ class Base
     virtual ~Base() {}
 
     /**
-     * Sets the message receive handler and associated argument
+     * Sets the owner of the transport
```

Comment:
Does this create circular references? I.e. before transport/raw was independent and transport was using transport/raw.

After this, transport/raw references transport and transport references transport/raw.

I think ideally individual modules should be independend. Could we maybe create transport/base (or core ... as much as I dislike generic names) that contains delegate definitions and callback usages?

Reply:
I am guessing gn may not allow you to do circular (there is a setting to force it, which is used because our intial code already had such references, however it should not be used for new code)